### ISSUE TITLE
Feature request: Split RMap component and context

### ISSUE THREAD

**Issue by andrejilderda** [lhapaipai/maplibre-react-components#3]
### Description

Hi @lhapaipai ,

This library looks very useful and I'm considering to use it for a project I'm working on. There's something I miss though.

Currently, the `RMap` component handles both the context for child components and the rendering of the map. This coupled approach can limit flexibility in more complex layouts when we need to access the map context outside of the immediate children of `RMap`, which is why I'd like to see those split into two separate parts:

1. An `RMapContext` provider
2. An `RMap` component for actual map rendering

The proposed Solution is to simply create a new `RMapContext` provider that can wrap a larger part of the application, and modify the `RMap` component to use this context rather than creating its own.

## Benefits
1. Increased flexibility in component structure and layout
2. Easier access to map-related data and functions throughout the app
3. Cleaner separation of concerns between context management and map rendering

Happy to hear your thoughts or any questions you might have!

### Example

```jsx
<RMapContext>
  {/* Other components that need access to the map */}
  <RMap>
    <ChildComponents />
  </RMap>
  {/* ... */}
</RMapContext>
```

**Comment by lhapaipai** [comment]
Hi @andrejilderda,
thanks for your feedback.
with pleasure, I love working on this library. I use it for my projects (personal and professional) and I wanted to share it. I want the library to remain lightweight without big dependencies but I have only implemented what I use so some features may be missing and there is no problem to add them

I will implement the feature in the next couple days, I probably ask your opinion or your review as soon as everything is ready because there are some side effects that hadn't occurred to me. 
- the fact that `useMap("some-id")` could be null
- re-render when it is set...
Have a good evening

**Comment by lhapaipai** [comment]
thanks for your review @andrejilderda. new features are available in v0.1.7
now it's your turn :smile:
let me know if you need any clarification or if some features are missing

**Comment by andrejilderda** [comment]
> I will implement the feature in the next couple of days.

Boom, 10 hours later and it’s done, awesome! 😃 Great to hear you enjoy working on this library. 

From my side, the migration from `@mapcomponents/react-maplibre` was seamless ([except for this small bug](https://github.com/lhapaipai/maplibre-react-components/issues/5)). Loving how lightweight this package is, and it’s great to see a solid Maplibre library for React!
